### PR TITLE
[VL-131] - Dependent checks send notification

### PR DESCRIPTION
**Pull request by kin0992**
Add the checks that depends on other API calls (e.g. PreLoad and UploadToS3)

#### How Has This Been Tested?

Unit tests

#### Types of changes

- [x] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/domain/checklist/sendPaymentNotificationChecklist.ts [hunk 1]
```diff
@@ -62,24 +58,24 @@ export const uploadToS3Check = {
...
-        P.and(hasRecipientDigitalDomicile),
-        P.and(hasPhysicalAddress),
-        P.and(hasRegisteredLetterAsPhysicalDocumentType),
-        P.and(hasRecipientPaymentCreditorTaxId),
-        P.and(hasRecipientPaymentNoticeCode),
-        P.and(hasSuccessfulResponse)
+        newNotificationRecordList,
+        RA.some(
+          pipe(
+            matchNewNotificationRecordCriteria,
```
Comment (datalek) [inline src/domain/checklist/sendPaymentNotificationChecklist.ts:74]: Just a suggestion, the choice is yours
```suggestion
            existsApiKey,
            P.and(hasRecipientTaxId),
            P.and(hasRecipientDigitalDomicile),
            P.and(hasPhysicalAddress),
            P.and(hasRegisteredLetterAsPhysicalDocumentType),
            P.and(hasRecipientPaymentCreditorTaxId),
            P.and(hasRecipientPaymentNoticeCode),
            P.and(hasSuccessfulResponse),
```

File: src/domain/NewNotificationRepository.ts [hunk 2]
```diff
@@ -61,6 +97,41 @@ export const makeNewNotificationResponse =
...
+export const matchNewNotificationRecordCriteria = pipe(
+  existsApiKey,
+  P.and(hasRecipientTaxId),
+  P.and(hasRecipientDigitalDomicile),
+  P.and(hasPhysicalAddress),
+  P.and(hasRegisteredLetterAsPhysicalDocumentType),
+  P.and(hasRecipientPaymentCreditorTaxId),
+  P.and(hasRecipientPaymentNoticeCode),
+  P.and(hasSuccessfulResponse)
+);
```
Comment (datalek) [inline src/domain/NewNotificationRepository.ts:109]: Remove it if you choose to move them into the [sendPaymentNotificationChecklist.ts](https://github.com/pagopa/pn-local-emulator/pull/67/files#diff-a94b02017c200eab4581cf751d371018a465dd8eecd987451d69de94d99e5a2c) file

File: src/domain/checklist/sendPaymentNotificationChecklist.ts (hunk 1 above)
Comment (kin0992) [reply src/domain/checklist/sendPaymentNotificationChecklist.ts]: Done!

File: src/domain/NewNotificationRepository.ts (hunk 2 above)
Comment (kin0992) [reply src/domain/NewNotificationRepository.ts]: Done!

**Review by datalek — APPROVED** [review]
🎩

**Pull request merged**